### PR TITLE
Fixes #2693 Fix system language change crash and support for language hot reload

### LINKED ISSUE #2693: FxR crashes after a configuration change
## Configuration

Firefox Reality version: 4ff1f5916302cb439d6d7bfff7573e14497545da
Firefox Reality build ID: eb9a1ec3e1d5ebfb37bb2686b134092b589d35a1

Hardware: 
HTC Vive Focus Plus - Android 7.1.2
HTC Vive Focus - Android 7.1.2
PicoVR (All)

## Steps to Reproduce

1. Open FxR
2. Exit to the dashboard
3. Go to the system settings and change the language
4. Open FxR

## Current Behavior

The app crashes after being relaunched.

## Expected Behavior

The app should open successfully.

## Context

During a normal Dashboard and app reopen flow, the lifecycle flow looks like:
`onPause` -> `onStop` -> (**DashBoard**) -> `onStart` -> `onResume`

But after a configuration change the flow is:
`onPause` -> `onStop` -> (**Configuration Change**) -> `onDestroy` -> `onCreate` -> **CRASH**

## Error Logs and Stack Traces

<details open>

```
java.lang.IllegalArgumentException
	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1020)
	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1330)
	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:608)
	at org.mozilla.vrbrowser.VRBrowserActivity.onDestroy(VRBrowserActivity.java:5)
	at android.app.Activity.performDestroy(Activity.java:7090)
	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1171)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4533)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4571)
	at android.app.ActivityThread.-wrap6(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1588)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:173)
	at android.app.ActivityThread.main(ActivityThread.java:6608)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:972)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:852)
```

</details>

### PR DESCRIPTION
**Pull request by keianhzo**
Fixes #2693 Support for config changes and language hot reload.

This fixes the related crash when changing the system language and also adds support for language hot reloading Fennec/Fenix style so no more restarts are required after changing languages.

There are a couple of things that I didn't add to this PR as it was getting too big already:
- [ ]  Refactor UI code to ViewModels to separate model/view which is very coupled right now. I started that for the PopUps view but there are still a lot of widgets to migrate. That would give us a better/easier way of saving/restoring UI states.
- [ ]  Add support for a default "Follow system language" as Fenix does so it's clearer for the user which language is being used.

Things to consider while testing:
- All widgets states restoration after a language change
- All widgets are correctly translated after a language change
- Languages changes both at a system and app levels

### CODE REVIEW

File: app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java [hunk 1]
```diff
@@ -1445,7 +1451,7 @@ public boolean canOpenNewWindow() {
     @Override
     public void openNewWindow(String uri) {
         WindowWidget newWindow = mWindows.addWindow();
-        if (newWindow != null) {
+        if (newWindow.getSession() != null) {
```
Comment (bluemarvin) [inline app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Even though `Windows.addWindow()` is marked as `@NonNull` for the return type, it actually can return `null`. So probably need to both fix the `addWindow()` annotation as well as revert this  change and make it:
```
    if ((newWindow != null) && (newWindow.getSession() != null)) {
        ...
    }
```

Comment (keianhzo) [reply app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java]: Fixed

**Comment by keianhzo** [comment]
Updated with support for ViewModel in the NavigationBar

**Comment by bluemarvin** [comment]
Lets land so QA can start testing.

**Pull request merged**